From ea0bf50d7a05df4733ae9d4f26f3520063ea419c Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 21 Jul 2025 14:50:50 +0200
Subject: [PATCH 5/7] Test EntityCombustByEntityEvent

---
 .../bukkit/listener/EventAbstractionListener.java         | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
index cb76b101..8db2f8cd 100644
--- a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
+++ b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
@@ -938,7 +938,10 @@ public void onEntityCombust(EntityCombustEvent event) {
                 // show as lit on the client consistently
                 return;
             }
-            Events.fireToCancel(event, new DamageEntityEvent(event, create(((EntityCombustByEntityEvent) event).getCombuster()), event.getEntity()));
+            event.getEntity().getScheduler().run(getPlugin(), scheduledTask -> {
+                Cause cause = create(((EntityCombustByEntityEvent) event).getCombuster());
+                Events.fireToCancel(event, new DamageEntityEvent(event, cause, event.getEntity()));
+            }, null);
         }
     }
 
@@ -987,7 +990,8 @@ public void onPlayerDropItem(PlayerDropItemEvent event) {
     @EventHandler(ignoreCancelled = true)
     public void onVehicleDamage(VehicleDamageEvent event) {
         Entity attacker = event.getAttacker();
-        Events.fireToCancel(event, new DamageEntityEvent(event, create(attacker), event.getVehicle()));
+        if (attacker == null) return;
+        attacker.getScheduler().run(getPlugin(), scheduledTask -> Events.fireToCancel(event, new DamageEntityEvent(event, create(attacker), event.getVehicle())), null);
     }
 
     //-------------------------------------------------------------------------
-- 
2.50.1.windows.1

