From 2007e68acee470efa29216cc084b5c92565fc6ec Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Thu, 28 Nov 2024 09:00:25 +0100
Subject: [PATCH 5/5] Test EntityCombustByEntityEvent

---
 .../bukkit/listener/EventAbstractionListener.java         | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
index 1ac26437..0bc8a334 100644
--- a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
+++ b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/EventAbstractionListener.java
@@ -925,7 +925,10 @@ public void onEntityCombust(EntityCombustEvent event) {
                 // show as lit on the client consistently
                 return;
             }
-            Events.fireToCancel(event, new DamageEntityEvent(event, create(((EntityCombustByEntityEvent) event).getCombuster()), event.getEntity()));
+            event.getEntity().getScheduler().run(getPlugin(), scheduledTask -> {
+                Cause cause = create(((EntityCombustByEntityEvent) event).getCombuster());
+                Events.fireToCancel(event, new DamageEntityEvent(event, cause, event.getEntity()));
+            }, null);
         }
     }
 
@@ -967,7 +970,8 @@ public void onPlayerDropItem(PlayerDropItemEvent event) {
     @EventHandler(ignoreCancelled = true)
     public void onVehicleDamage(VehicleDamageEvent event) {
         Entity attacker = event.getAttacker();
-        Events.fireToCancel(event, new DamageEntityEvent(event, create(attacker), event.getVehicle()));
+        if (attacker == null) return;
+        attacker.getScheduler().run(getPlugin(), scheduledTask -> Events.fireToCancel(event, new DamageEntityEvent(event, create(attacker), event.getVehicle())), null);
     }
 
     //-------------------------------------------------------------------------
-- 
2.45.1.windows.1

