From 179e24287a13f256c942d093a6e01330d5e7c2a4 Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 9 Feb 2026 00:25:16 +0100
Subject: [PATCH 7/8] Fix https github.com Euphillya WorldGuard Folia issu

---
 .../listener/WorldGuardPlayerListener.java    | 23 +++++++++++++++++++
 1 file changed, 23 insertions(+)

diff --git a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/WorldGuardPlayerListener.java b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/WorldGuardPlayerListener.java
index 1656ab32..b44b5e49 100644
--- a/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/WorldGuardPlayerListener.java
+++ b/worldguard-bukkit/src/main/java/com/sk89q/worldguard/bukkit/listener/WorldGuardPlayerListener.java
@@ -43,6 +43,7 @@
 import org.bukkit.Material;
 import org.bukkit.World;
 import org.bukkit.block.Block;
+import org.bukkit.entity.EnderPearl;
 import org.bukkit.entity.Entity;
 import org.bukkit.entity.Player;
 import org.bukkit.event.Event;
@@ -444,4 +445,26 @@ public void onPlayerCommandPreprocess(PlayerCommandPreprocessEvent event) {
             }
         }
     }
+
+    // Fix ENDERPEARL folia
+    @EventHandler(priority = EventPriority.LOW, ignoreCancelled = true)
+    public void onPlayerLaunchProjectileEvent(final com.destroystokyo.paper.event.player.PlayerLaunchProjectileEvent event) {
+        if (event.getProjectile() instanceof EnderPearl pearl) {
+            if (pearl.getShooter() instanceof Player player) {
+                LocalPlayer localPlayer = getPlugin().wrapPlayer(player);
+                if (!WorldGuard.getInstance().getPlatform().getSessionManager().hasBypass(localPlayer, localPlayer.getWorld())) {
+                    RegionQuery query = WorldGuard.getInstance().getPlatform().getRegionContainer().createQuery();
+                    ApplicableRegionSet set = query.getApplicableRegions(BukkitAdapter.adapt(pearl.getLocation()));
+                    if (!set.testState(localPlayer, Flags.ENDERPEARL)) {
+                        event.setCancelled(true);
+                        String message = set.queryValue(localPlayer, Flags.DENY_MESSAGE);
+                        if (message != null && !message.isEmpty()) {
+                            player.sendMessage(message);
+                        }
+                    }
+                }
+            }
+        }
+    }
+
 }
-- 
2.52.0.windows.1

